### DISCUSSION TITLE
Curious about use cases

### DISCUSSION THREAD

**Discussion by highfestiva** [highfestiva/finplot#92] [Show and tell]
Many feature requests I get indicate users of this library seem to use it wrapped in a tiny application for manual inspection of various things. So let's do a poll. What is your main purpose of using finplot? Pick one choice (for stats), but feel free to elaborate.

1. Backtesting of automated trading using technical analysis.
2. Backtesting of automated trading using ANN.
3. Manually exploring, using finplot as a drop-in replacement for other python libs.
4. Manually exploring, replacing TradingView or similar online plotting services.
5. Using finplot as a free alternative for displaying e.g. volume profile.
6. Generating images for web, social media, alarms, etc.
7. Other.

I'm also curious of what instruments you're into, what indicators you use and if you'd be interested in a fully fledged example application — and if so for what reason and purpose?

**Comment by highfestiva** [comment]
So I'll start. [1] for BTCUSDTPERP. I don't use standard indicators. Visualizing 1M candles or so.

**Comment by beglitis** [comment]
I am quite excited about this library so I'll chime in. 3-6 are definitely within scope. 1 and 2 still thinking how can I go about it. However I have been intrigued by the live updating of the plots from streamed data (e.g. like in the example in the repo with data from BitMex). So two use cases I am developing right now. One is a live PnL plot. Get live prices in, display them in real time and overlay on top the portfolio's PnL. The second case is as the front-end of an execution engine. Get live information in (prices, volumes etc), draw the signals on top and the buys and sells executed by the engine.

**Reply by grantxx** [reply]
Definitely number 1 for me. I would like to run backtests and plot the results on the chart using little arrows showing entry and exit prices. 
I trade ES futures and 6E 
I rely on my own basic indicators such as round number levels, average daily range projections, and text on the chart showing last x sessions closes higher/lower, average points made etc. I can provide a screenshot if necessary,

**Reply by highfestiva** [reply]
Do you guys use leverage at all? I've both gained a lot, and lost almost as much using high leverage. Now I'm using very low leverage.

**Reply by grantxx** [reply]
Same. I prefer not using high leverage. I trade small and consistent as Id prefer to be here in the long run. What timeframe do you trade?

**Reply by highfestiva** [reply]
I've dabbled with 1m to 15m, and at some point 1h. Mostly 1m, which allows one to get on the (correct) bandwagon quickly, or fail fast. Mostly the latter for me though. ;)

**Comment by finnstadel** [comment]
I'd say 1 and 4. Started out as a command line tool, but now growing toward a more interactive version. Trying to develop new indicators based on signal processing techniques -  maybe moving towards/including deep learning algorithms. If nothing else, then I'm the one that goes through the deep learning process... ha ha :-)
In the old days, I would have written everything in C/C++ or perhaps matlab (sorry octave ;-) ), but there are so many high level libraries in python these days, so that's where I'm at ;-)
Cheers !

**Reply by highfestiva** [reply]
Sounds like me, but was never able to make much sense of the ML. But hopefully my homebrew indicators are sufficient. :)

**Reply by AlexStox** [reply]
It's hard to choose only one :) But if the only one choice is important for stat, I would choose n.4. 

But to be honest, my goal is a mix of 1, 3, 4, 5. 

I like your lib very much! I've been trading different tools, mostly futures. I don't like standard and well-known indicators and technical analysis approaches. So I'm trying to find some inefficiencies in a market. It was hard to do manually for years, even Excel was not the best solution :) That's why I've started learning Python and Pandas since last autumn. 

I'm trying to code my own indicators and filters just to help me withdraw some routine work that was previously made by hand :) Although I have lots of ideas to analyze markets, and these ideas are still waiting for me to code better :) 

Your library helped by saving a lot of time!

In the nearest future I'd like to check whether it possible to show a multi-chart plot for different stocks and futures (symbols) in one output chart window?

**Comment by workingshaw** [comment]
I'd say my use case is manual backtesting / free alternative to software like forex tester.

**Comment by Camelket** [comment]
For me 4 and 6 (generating charts of trades automatically from tradelogs).
Currently trying to get live plotting to work, so I can complement my brokers trading app (costum/better looking indicators).
Into equity options/ equity at the moment using volume profile and VWAP.

**Comment by forna** [comment]
7. Other.
I am developing an application to aid me with entering trades. It should create alerts when some technical analysis conditions are met. 
I mainly trade Options on the NASDAQ. I can find a lot of tools/screeners for Stock based conditions (e.g. stockfetcher) but almost nothing for Options. The finplot Charts will help me to fine-tune the alert rules.

**Comment by Resumane** [comment]
So far one actual use case other than playing around and learning stuff: plotting price from different exchanges together to visualize the spread (obscure exchanges that are not in tradingview) and do various operations on this prices (currency conversion).

**Reply by highfestiva** [reply]
I've never tried arbitraging myself, does it seem profitable for a layman? How fast does that transaction need to be in order to earn money?
